### PR TITLE
enhanced error details

### PR DESCRIPTION
**Pull request by trygve-lie**
This PR is a small enhancement of the carrier for details of an error. These details can as is be anything, but they are intended to hold more descriptive details on an error than a stackTrace. These details are more in line of telling exactly what is wrong (ex; parameter "foo" is empty, should be "xxxx") and are intended to be used in a UI for giving an end user / or a service exact info of what is failing.

Its also now possible to pass in a validation error object from [Joi](https://github.com/hapijs/joi) and all its validations details will be kept as is. 

Ex; if a API is returning JSON and a validation is failing (in this example the parameter "a" wants a string, but a number was passed in), the returned http error will look like this:

```json
{                                                                 
  "details": [                                                    
    {                                                             
      "context": {                                                
        "key": "a",
        "value": 10                                               
      },                                                        
      "message": "'a' must be a string",
      "path": "a",
      "type": "string.base"                                       
    }                                                             
  ],
  "message": "Bad Request",
  "statusCode": 400                                               
}  
```

### CODE REVIEW

File: lib/error.js [hunk 1]
```diff
@@ -16,14 +16,17 @@ const isObject = require('lodash.isobject');
...
+    this.details = [];
 
     if (isObject(problem)) {
-        this.problem = problem;
+        this.details.push(problem);
     }
 
     if (isError(message)) {
         this.message = message.message;
+        if (Array.isArray(message.details)) {
```
Comment (magnli10) [inline lib/error.js]: Details can be either array or string?

Comment (magnli10) [inline lib/error.js]: Update param annotation if message is not only string

Comment (trygve-lie) [inline lib/error.js]: I'm not quite following you. Please elaborate.

Comment (magnli10) [inline lib/error.js]: This, message can be other than string :) https://github.com/amedia/node-error-obj/blob/joi/lib/error.js#L12

**Comment by magnli10** [comment]
Looks good 👍

File: lib/error.js (hunk 1 above)
Comment (trygve-lie) [inline lib/error.js]: Ah. Then I'm with you :)

**Pull request merged**